### DISCUSSION TITLE
A possible bug for mac users with JUCE(C++)

### DISCUSSION THREAD

**Discussion by caoster** [sist-cs120/project-wiki#30] [Show and tell]
This is a solved bug.
I use an (Intel) mac and here is the problem I face:
Sometimes (actually most of the time: 19 out of 20), my program gets stuck in `setAudioChannels(1, 1);` and will never return from there, no matter how I set my CMakeList, what compiler I use or which JUCE library I download (version).
After struggling through project1, I accidentally find that the program works very well when I don't use "MacBook Pro扬声器" as my speaker. This is not a big deal in project2/3/4, as we don't use our own speaker anyway, but it wasn't enjoyable in project 1: I spent nearly an hour when performing checks, and I really appreciate TAs for their patience.

Here is my solution:
Find the app "音频MIDI设置" on your mac, click "+", then "创建聚合设备", double click on its name to make it ASCII only (simply avoid possible troubles), put "MacBook Pro扬声器" into it, and set your brand new virtual device as output.

You can also use another 3.5mm/USB speaker instead of the built-in one, but you might miss the performance of your expensive built-in devices.

BTW, C++ multithreading with JUCE is easy and fast! I strongly recommend you try this out.

**Comment by yangfengxu12** [comment]
Thank you for sharing.
